tests/device_tests/test_session_id_and_passphrase.py

```diff
@@ -0,0 +1,241 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2019 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import messages
+from trezorlib.tools import parse_path
+
+
+def _get_xpub(client, passphrase):
```

Comment:
there should be an explanatory comment saying that all the calls are raw here (as opposed to using library functions) to avoid trezorlib's implicit session management (clearing session after settings, hidden handling of session ids, etc)

also there should not be `setup_client` decorators anywhere, as empty setup_client should never be needed